### PR TITLE
Feature/SIE-103 redesign image stimuli observer

### PR DESCRIPTION
**Pull request by theodore3131**
In this PR:
- Redesigned the process of observing image stimuli generation from Pub/Sub message to firestore document
- Fix the uploadImage path issue.

For frontend:  use `useEffect()` to call for the observe function to get the stimuli-image URLs.

### CODE REVIEW

File: src/firebase/api/gcp-utils.js [hunk 1]
```diff
@@ -26,60 +26,59 @@ export const uploadSelfImage = async (userId, experimentId, image) => {
  */
 const uploadImageToStorage = async (
   userId, experimentId, image, bucket) => {
-  const imagePath = `${userId}-${experimentId}/${image.name}`;
+  const imageName = `${userId}-${experimentId}.${image.name.split('.').pop()}`;
```
Comment (quanganhhoang) [inline src/firebase/api/gcp-utils.js:29]: javascript is interesting

File: src/firebase/api/gcp-utils.js [hunk 2]
```diff
@@ -89,30 +88,18 @@ export const observeStimuliCompletion =
  */
 const getFileUrlsFromBucket = async (userId, experimentId) => {
```
Comment (quanganhhoang) [inline src/firebase/api/gcp-utils.js:89]: let's have a generic function getFileUrlsFromBucket where `userId` and `experimentId` are required and maybe a filter as optional? So we can use this to filter only images or something, then we can have another method called `getSieStimuliFromBucket`.

File: src/firebase/api/gcp-utils.js [hunk 3]
```diff
@@ -26,60 +26,59 @@ export const uploadSelfImage = async (userId, experimentId, image) => {
...
       message: `user not authenticated ${error.code}`,
-      data: {},
+      data: null,
     };
   });
 };
 
-
 /**
  * Subscribes to pubsub to signal stimuli generation completion
```
Comment (quanganhhoang) [inline src/firebase/api/gcp-utils.js:49]: Let's update these comments

**Review by quanganhhoang — CHANGES_REQUESTED** [review]
Awesome work! I left a few comments.

File: src/firebase/api/gcp-utils.js (hunk 2 above)
Comment (theodore3131) [reply src/firebase/api/gcp-utils.js:89]: sounds good to me, will update!

File: src/firebase/api/gcp-utils.js [hunk 4]
```diff
@@ -26,93 +26,115 @@ export const uploadSelfImage = async (userId, experimentId, image) => {
...
-        });
-      } else {
-        // “face_missing”, “failed”
-        // call setError() for image processing status, display on frontend
-        errorHandler(status);
-      }
-    });
+          await getSieStimuliFromBucket(userId, experimentId)
+            .then((results) => {
+              console.log(results);
```
Comment (quanganhhoang) [inline src/firebase/api/gcp-utils.js:66]: remove console.log

File: src/firebase/api/gcp-utils.js [hunk 5]
```diff
@@ -26,93 +26,115 @@ export const uploadSelfImage = async (userId, experimentId, image) => {
...
+              const status = results.status;
+              if (status === StatusCodes.OK) {
+              // successfully get all image urls
+                // return results;
```
Comment (quanganhhoang) [inline src/firebase/api/gcp-utils.js:70]: pls remove comments

File: src/firebase/api/gcp-utils.js [hunk 6]
```diff
@@ -26,93 +26,115 @@ export const uploadSelfImage = async (userId, experimentId, image) => {
...
+export const unsubscribe = firestore.collection(firestoreCollections.USER)
+  .onSnapshot(() => {
+    return;
+  });
+
+
 /**
- * Get all file urls from bucket to display in img tags
+ * Get all stimuli image urls from bucket to display in img tags.
+ * @param {*} userId user id
```
Comment (quanganhhoang) [inline src/firebase/api/gcp-utils.js:97]: string

**Review by quanganhhoang — APPROVED** [review]
LGTM! Just a few nits before you merge

File: src/firebase/api/gcp-utils.js (hunk 4 above)
Comment (theodore3131) [reply src/firebase/api/gcp-utils.js]: Thank you! My bad for not run a lint check.

**Pull request merged**